contracts/SpokePool.sol

```diff
@@ -698,20 +700,26 @@ abstract contract SpokePool is SpokePoolInterface, Testable, Lockable, MultiCall
     ) internal view {
         // A depositor can request to speed up an un-relayed deposit by signing a hash containing the relayer
         // fee % to update to and information uniquely identifying the deposit to relay. This information ensures
-        // that this signature cannot be re-used for other deposits. The version string is included as a precaution
-        // in case this contract is upgraded.
-        // Note: we use encode instead of encodePacked because it is more secure, more in the "warning" section
-        // here: https://docs.soliditylang.org/en/v0.8.11/abi-spec.html#non-standard-packed-mode
-        bytes32 expectedDepositorMessageHash = keccak256(
-            abi.encode("ACROSS-V2-FEE-2.0", newRelayerFeePct, depositId, originChainId)
+        // that this signature cannot be re-used for other deposits.
+        // Note: We use the EIP-712 (https://eips.ethereum.org/EIPS/eip-712) standard for hashing and signing typed data.
+        // Specifically, we use the version of the encoding known as "v4", as implemented by the JSON RPC method
+        // `eth_signedTypedDataV4` in MetaMask (https://docs.metamask.io/guide/signing-data.html).
+        bytes32 expectedTypedDataV4Hash = _hashTypedDataV4(
+            // EIP-712 compliant hash struct: https://eips.ethereum.org/EIPS/eip-712#definition-of-hashstruct
+            keccak256(
+                abi.encode(
+                    keccak256(
+                        "UpdateRelayerFeeMessage(uint64 newRelayerFeePct,uint32 depositId,uint256 originChainId)"
+                    ),
+                    newRelayerFeePct,
+                    depositId,
+                    originChainId
+                )
+            ),
```

Comment:
Yea, I agree that this is a bit nested. But tbh I don't know what to put in local variables because this hash structure is defined like that in the EIP. I could do 
```solidity
bytes32 hashStruct = keccak256(...);
bytes32 expectedTypedDataV4Hash = _hashTypedDataV4(hashStruct);
```
but not sure if this makes it really more readable?

Reply:
Is there a gas difference if you change the syntax? Can run the test like so `REPORT_GAS=true hardhat test`

Reply:
@nicholaspai Thanks for the suggestion. Yes, with the additional local variable the methods `speedUpDeposit` and `fillRelayWithUpdatedFee` cost 10 gas more.

Reply:
this is an interesting case where smart contract engineering is different from off-chain engineering. Temp variables increase readability but incur a real cost! In this case I'd opt for the reduction in readability to save gas